src/App.tsx

```diff
@@ -6,29 +6,32 @@ import { SignUp } from "./component/SignUp";
 import { BrowserRouter as Router, Switch, Route } from "react-router-dom";
 import DataGridDemo from "./component/TableView";
 import { Analytics } from "./screens/Analytics";
+import { AuthProvider } from "./context/AuthContext";
 
 function App() {
   return (
     <div className="App">
       <header className="App-header">
-        <Router>
-          <Switch>
-            <Route path="/sign-up">
-              <SignUp />
-            </Route>
-            <Route path="/login">
-              <Login />
-            </Route>
-            <Route path="/analytic">
-              <Analytics />
-            </Route>
-            <Route path="/">
-              {/* "Your past streams" could be replaced with {regular or twitch username}'s past streams */}
-              <p>Your past streams</p>
-              <DataGridDemo />
-            </Route>
-          </Switch>
-        </Router>
+        <AuthProvider>
```

Comment:
I was wondering about where this should be, for example if we had to use the provider on this component it might make sense to wrap `App` with the provider? (For example for route guards)